redis-utils/src/main/java/org/ballerinalang/redis/actions/AbstractRedisAction.java

```diff
@@ -1522,12 +1460,12 @@ protected BValueArray hKeys(String key, RedisDataSource<String, String> redisDat
         }
     }
 
-    private boolean isClusterConnection(RedisDataSource redisDataSource) {
+    private static boolean isClusterConnection(RedisDataSource redisDataSource) {
         return redisDataSource.isClusterConnection();
     }
 
-    private BValueArray createBStringArrayFromSet(Set<String> set) {
-        BValueArray bStringArray = new BValueArray(new String[set.size()]);
+    private static BArray createBStringArrayFromSet(Set<String> set) {
+        BArray bStringArray = new ArrayValueImpl(new String[set.size()]);
         int i = 0;
         for (String item : set) {
             bStringArray.add(i, item);
```

Comment:
Shall we use `bStringArray.append()` instead of `bStringArray.add()` ?
We can get rid of the counter variable.

Reply:
Updated in a6ce8d8